File: mobile_app/lib/classroom/model/classroom.dart
class Classroom {
  final String id;
  final String name;
  final String? faculty;
  final int startYear;
  final int endYear;
  final DateTime createdAt;
  final int studentCount;
  final bool isActive;

  Classroom({
    required this.id,
    required this.name,
    this.faculty,
    required this.startYear,
    required this.endYear,
    required this.createdAt,
    required this.studentCount,
    required this.isActive,
  });

  factory Classroom.fromMap(Map<String, dynamic> map) {
    return Classroom(
      id: map['id'],
      name: map['name'],
      faculty: map['faculty'] ?? '',
      startYear: (map['start_year']),
      endYear: map['end_year'],
      createdAt: DateTime.parse(map['created_at']).toLocal(),
      studentCount: map['student_count'],
      isActive: map['is_active'],
    );
  }
}

File: admin_web/src/models/classroom.ts
export class Classroom {
  id: string | null;
  name: string;
  createdAt: Date;
  faculty: string;
  startYear: number;
  endYear: number;
  isActive: boolean; 
  constructor(
    id: string | null,
    name: string,
    createdAt: Date,
    faculty: string,
    startYear: number,
    endYear: number,
    isActive: boolean
  ) {
    this.id = id;
    this.name = name;
    this.createdAt = createdAt;
    this.faculty = faculty;
    this.startYear = startYear;
    this.endYear = endYear;
    this.isActive = isActive;
  }

  static fromMap(map: any): Classroom {
    return new Classroom(
      map.id,
      map.name,
      new Date(map.created_at),
      map.faculty,
      map.start_year,
      map.end_year,
      map.is_active   
    );
  }

  toMap() {
    return {
      id: this.id,
      name: this.name,
      created_at: this.createdAt.toISOString(),
      faculty: this.faculty,
      start_year: this.startYear,
      end_year: this.endYear,
      is_active: this.isActive,  
    };
  }
}